### DISCUSSION TITLE
Add protection against volume and file mount deletion to prevent /data and other critical directories from being deleted.

### DISCUSSION THREAD

**Discussion by peaklabs-dev** [coollabsio/coolify#4010] [Improvement Requests]
- Currently, if you mount `/data` as the directory mount in the container, then delete the directory mount and select `delete on server`, `/data` is nuked. We need to prevent this somehow, as deleting `/data` will break Coolify.

**Comment by theDanielJLewis** [comment]
YIKES! I don't know how this was possible, but I think I just messed up my server because of this (and some mislabeled things).

I added a directory mount (not really seeing any difference between mounting directories and volumes—especially since the documentation doesn't explain them). That gave me two delete buttons "delete" and "delete directory." Clicking "delete directory" warned me that doing so would delete all that folder's contents from my server. So I clicked "delete" and proceeded … which deleted all that folder's contents from my server!

And since I was trying to do something with `/` and `/sys` (for cAdvisor), it ruined my server.

Good thing I had backups!

But that should not have been possible to do, especially with the button I clicked.

**Comment by arthrod** [comment]
Hey guys,

Actually answering to issue #2410 

In my case, it happened (a few times) when I was running as a non-root user.  The container runs as www-data (UID 9999) but the SSH keys directory had different ownership.

## The Fix

1. First, I checked my container ID:
```bash
docker ps | grep coolify
```

2. Then I jumped into the container as root (no sudo inside):
```bash
sudo docker exec -it --user=root [container-id] sh
```

3. And fixed those permissions with (can't be too loose otherwise ssh dislikes):
```bash
chown -R 9999:www-data /var/www/html/storage/app/ssh && chmod -R 770 /var/www/html/storage/app/ssh && chmod 660 /var/www/html/storage/app/ssh/keys/*
```

Ta-dah!

Thanks!

**Reply by thewasta** [reply]
This gave me a permissions error. SSH keys should have lower permissions, so I changed it to 600 and everything was fixed.
```
chown -R 9999:www-data /var/www/html/storage/app/ssh && chmod -R 770 /var/www/html/storage/app/ssh && chmod 600 /var/www/html/storage/app/ssh/keys/*
```
